src/main/java/io/supertokens/webserver/api/dashboard/DashboardUserAPI.java

```diff
@@ -0,0 +1,207 @@
+package io.supertokens.webserver.api.dashboard;
+
+import java.io.IOException;
+import java.io.Serial;
+
+import com.google.gson.JsonObject;
+
+import io.supertokens.Main;
+import io.supertokens.dashboard.Dashboard;
+import io.supertokens.pluginInterface.RECIPE_ID;
+import io.supertokens.pluginInterface.dashboard.exceptions.DuplicateEmailException;
+import io.supertokens.pluginInterface.exceptions.StorageQueryException;
+import io.supertokens.pluginInterface.exceptions.StorageTransactionLogicException;
+import io.supertokens.webserver.InputParser;
+import io.supertokens.webserver.WebserverAPI;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+
+public class DashboardUserAPI extends WebserverAPI {
+
+    @Serial
+    private static final long serialVersionUID = -3243962619116144573L;
+
+    public DashboardUserAPI(Main main) {
+        super(main, RECIPE_ID.DASHBOARD.toString());
+    }
+
+    @Override
+    public String getPath() {
+        return "/recipe/dashboard/user";
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws IOException, ServletException {
+
+        try {
+
+            if (!Dashboard.isFeatureFlagEnabledOrUserCountIsUnderThreshold(main)) {
```

Comment:
The `isFeatureFlagEnabledOrUserCountIsUnderThreshold` should be named to something more generic. This name wont work and will require refactor if the condition for considering someone as a paid dashboard user changes in the future